vertical-pod-autoscaler/pkg/recommender/input/history/history_provider.go

```diff
@@ -34,30 +35,68 @@ import (
 	metrics_recommender "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/utils/metrics/recommender"
 )
 
-// PrometheusBasicAuthTransport contains the username and password of prometheus server
+// PrometheusBasicAuthTransport injects basic auth headers into HTTP requests.
 type PrometheusBasicAuthTransport struct {
 	Username string
 	Password string
+	Base     http.RoundTripper
 }
 
 // RoundTrip function injects the username and password in the request's basic auth header
 func (t *PrometheusBasicAuthTransport) RoundTrip(req *http.Request) (*http.Response, error) {
-	req.SetBasicAuth(t.Username, t.Password)
-	return http.DefaultTransport.RoundTrip(req)
+	// Use default transport if none specified
+	rt := t.Base
+	if rt == nil {
+		rt = http.DefaultTransport
+	}
+
+	// Clone the request to avoid modifying the original
```

Comment:
This comment says what the code is doing, but not why the code is doing it. Could you add a comment linking to the best practise?